src/components/Tag.js

```diff
@@ -0,0 +1,75 @@
+import { Button, IconButton } from "@chakra-ui/react";
+import { useState } from "react";
+import Icon from "./Icon";
+
+const Tag = ({
+  label,
+  textColorUnselected,
+  textColorSelected,
+  tagColorSelected,
+  tagColorUnselected,
+  sourceSelected,
+  sourceUnselected,
+  isCategory,
+}) => {
+  const [selected, setSelected] = useState(false);
+
+  const handleClick = () => {
+    setSelected(!selected);
+  };
+
+  return (
+    <>
+      {!selected && sourceUnselected && (
+        <IconButton
+          aria-label=""
+          borderRadius="20px"
+          isActive={true}
+          _active={{
+            bg: tagColorUnselected,
+          }}
+          icon={<Icon circle={false} source={sourceUnselected} />}
+          onClick={handleClick}
+        />
+      )}
+      {!selected && !sourceUnselected && (
```

Comment:
This is really messy/hard to read. Think about how you can organize this better using ternarys and functons